src/components/Card.js

```diff
@@ -1,41 +1,82 @@
 class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { _id, name, link, isLiked },
+    cardSelector,
+    handleImageClick,
+    handleConfirmDelete,
+    handleCardLike,
+    handleCardDisLike
+  ) {
+    this._id = _id;
     this._name = name;
     this._link = link;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleConfirmDelete = handleConfirmDelete;
+    this._handleCardLike = handleCardLike;
+    this._isLiked = isLiked;
+    this._handleCardDisLike = handleCardDisLike;
   }
 
   _setEventListeners() {
     //".card__like-button"
     this._element
       .querySelector(".card__like-button")
-      .addEventListener("click", () => {
-        this._handleLikeButton();
+      .addEventListener("click", (e) => {
+        e.preventDefault();
+
+        if (this._isLiked) {
+          this._handleCardDisLike(this);
+          this._likeButton.classList.remove("card__like-button_active");
```

Comment:
* You haven't corrected my remark https://github.com/Agent610/se_project_aroundtheus/pull/1#discussion_r1779423531  You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.